### DISCUSSION TITLE
Unable to start fspctl

### DISCUSSION THREAD

**Discussion by ocxjuergen** [cspiel1/fspctl#7] [Q&A]
Hello! I´ve tried to install fspctl on the actual Raspian image with cmake.
Everything went fine and compiles without any errors.

But when I start fspctl (In Terminal: sudo ./fspctl) I get an error that fspctl cannot write the Config file to the root folder?
Can you Please help me out with this?

Maybe you can give me/us a detailed installation instruction?
Thank You

EDIT: Now i’m getting it to run - I created a .config Folder in the root source

**Comment by cspiel1** [comment]
Right, the .config folder has to be created.

**Reply by ocxjuergen** [reply]
And do I also have to write in the config file  the adresses for publish0 ->  1470 for ac input voltage and so on?

**Comment by cspiel1** [comment]
Example:
```
[alarm@fspctl ~]$ cat .config/.fspctl
ttydev     /dev/ttyWR
stopbits   1
slaveid    2
mqtthost   <host-address>
mqttport   <port>
mqttuser   <user>
mqttpass   <passw>
capath   /etc/ssl/certs
publish0   230 -1 pv/inverter/battery/current
publish1   904 8 pv/inverter/battery/power_direction
publish2   226 -1 pv/inverter/battery/capacity
publish3   188 -1 pv/inverter/battery/voltage
publish4   228 -1 pv/inverter/solar_input1/power
publish5   232 -1 pv/inverter/solar_input2/power
publish6   377 -1 pv/inverter/AC_input/power
```

**Reply by cspiel1** [reply]
I think you'll have to look in the code for details.

**Reply by ocxjuergen** [reply]
Alright thanks, that helps me a lot!

**Reply by ocxjuergen** [reply]
So im struggeling now with the modbus card - everytime i get modbus connection timeout - bitrate at the dip switches at  P17 card is 19200 8n1 Slave id 2 -> do i have to turn the card in „server mode“ or just let it as default? 

Also I‘ve tried 2 different modbus adapter (Ch340 and FT232L)- also I tried to change A and B lines.. always modbus connection timeout

Hope you can help me out - Thanks

**Reply by cspiel1** [reply]
Sorry, I am not able currently. I have not more than the code and the manuals. You have to study these carefully.
